### DISCUSSION TITLE
How to upload a single file using the Upload scalar?

### DISCUSSION THREAD

**Discussion by sanchesfrolov** [graphql-java-kickstart/graphql-java-tools#376] [Q&A]
Hello team !
I am pretty new with GraphQL and I need help with uploading a single file.
I was following [the spec](https://github.com/jaydenseric/graphql-multipart-request-spec) and related issues from the repo: 
[https://github.com/graphql-java-kickstart/graphql-java-tools/issues/240](url), 
[https://github.com/graphql-java-kickstart/graphql-java-tools/issues/288](url)

I have following dependencies in my gradle script:
    implementation "com.graphql-java-kickstart:graphql-kickstart-spring-boot-starter-webflux:6.0.1"
    implementation "com.graphql-java-kickstart:graphql-kickstart-spring-boot-autoconfigure-tools:6.0.1"

and I added following implementation:

> scalar Upload
>type Mutation {
testSingleFileUpload(file: Upload): UploadResult
}

 inside schema.graphql.

> @bean
> GraphQLScalarType upload()
> {
>         return ApolloScalars.Upload;
> }

 inside my configuration.

> public UploadResult testMultiFilesUpload(Part part, DataFetchingEnvironment environment) {
>    Part part1 = environment.getArgument("file");  
>     ...  
>  }

Inside mutation resolver

In order to check it I used Altair GraphQL Client(chrome-extension).
So when I post my request it eventually comes to > > >graphql.kickstart.spring.AbstractGraphQLController.graphqlPOST method 

and 

> ResponseStatusException(HttpStatus.UNPROCESSABLE_ENTITY, "Could not process GraphQL request")

 is thrown because the request has 

> multipart/form-data; 

as a content type, but the method waits for **_application/json_** or **_application/graphql_** 

I think I missed something important in my implementation and just need more detail how I could set up all needed environment for file uploading.

**Comment by ayush11-11** [comment]
Bad POST multipart request: no part named "graphql" or "query" getting this while sending response through potman

**Comment by LeKuetebi** [comment]
Hey sanchesfrolov. By any chance did you get a solution for this problem? I'm facing the same issue.
